tests/foreman/ui_airgun/test_hostcollection.py

```diff
@@ -635,3 +702,86 @@ def test_negative_hosts_limit(session, module_org, module_loc):
         assert "cannot have more than 1 host(s) associated with host collection '{}'".format(
             hc_name
         ) in str(context.value)
+
+
+@tier3
+@upgrade
+def test_positive_install_module_stream(
+        session, module_org, vm_content_hosts_module_stream, vm_host_collection_module_stream):
+    """Install a module-stream to hosts inside host collection remotely
+
+    :id: e5d882e0-3520-4cb6-8629-ef4c18692868
+
+    :Steps:
+        1. Running dnf upload profile to sync module streams from hosts to Satellite
+        2. Navigating to host_collection
+        3. Installing the module stream walrus
+        4. Verifying that remote job get passed
+        5. Verifying that package get installed
+
+    :expectedresults: Module-Stream was successfully installed on all the hosts
+        in host collection
+
+    :CaseLevel: System
+    """
+    stream_version = "0"
+    with session:
+        _run_remote_command_on_content_hosts(
+            'dnf -y upload-profile',
+            vm_content_hosts_module_stream
+        )
+        session.organization.select(org_name=module_org.name)
+        result = session.hostcollection.manage_module_streams(
+            vm_host_collection_module_stream.name,
+            action_type="Install",
+            module_name=FAKE_3_CUSTOM_PACKAGE_NAME,
+            stream_version=stream_version
+        )
+
+        assert result['overview']['job_status'] == 'Success'
+        assert result['overview']['job_status_progress'] == '100%'
+        assert int(result['overview']['total_hosts']) == 2
+        assert _is_package_installed(vm_content_hosts_module_stream, FAKE_3_CUSTOM_PACKAGE)
+
+
+@tier3
+@upgrade
+def test_positive_install_modular_errata(
+        session, module_org, vm_content_hosts_module_stream, vm_host_collection_module_stream):
+    """Install Modular Errata generated from module streams.
+
+    :id: 8d6fb447-af86-4084-a147-7910f0cecdef
+
+    :Steps:
+        1. Generated errata by installing and downgrading the Module Stream Kangaroo
```

Comment:
have a query , can we not install only downgraded module stream `Kangaroo` and upload new profile to generate installable  erratas  ?

Reply:
Modular errata can only be seen if the dependent package of module stream is available with errata. If we install any new profile by default it will be installed at the latest version of that profile only. Consider an example if you see dnf module list nodejs with nodejs 6, 7.  

But installing nodejs 6 or with any profile won't show you any errata if you considering there is a more updated version of 7. In this case, errata have only been if the nodejs 6 modules dependent packages are older with respect to the repository 6.1 i.e. https://docs.fedoraproject.org/en-US/modularity/architecture/consuming/upgrade-paths/  (in case if understood your question correctly)

Currently, I don't know any other way we can generate this errata, this what I understood from the developer. Even if it could, I still think not possible with at least this custom repository.

Reply:
It works in below way , please give a try :
~~~
# dnf module install kangaroo:0:20180704111719
Updating Subscription Management repositories.
swid_zoo_repo                                                                                                                                 56 kB/s | 3.3 kB     00:00    
Dependencies resolved.
=============================================================================================================================================================================
 Package                            Arch                             Version                         Repository                                                         Size
=============================================================================================================================================================================
Installing group/module packages:
 kangaroo                           noarch                           0.2-1                           SWID_Testing_SWID_Product_swid_zoo_repo                           1.8 k

Transaction Summary
=============================================================================================================================================================================
Install  1 Package

Total download size: 1.8 k
Installed size: 42  
Is this ok [y/N]:

 dnf -y upload-profile
Updating Subscription Management repositories.
Package profile updates
        status: 1
        updates: []
        exceptions: 

~~~

Reply:
@vijay8451 works, as you told earlier profile. I was looking into the profile. But this works.